cartoframes/context.py

```diff
@@ -1105,3 +1120,47 @@ def _df2pg_schema(dataframe, pgcolnames):
     if 'the_geom' in pgcolnames:
         return '"the_geom", ' + schema
     return schema
+
+
+class BatchJobStatus(object):
+    """Batch SQL API job status. Read more at `Batch SQL API docs
+    <https://carto.com/docs/carto-engine/sql-api/batch-queries/>`__.
+
+    Example:
+
+        .. code:: python
+
+            import time
+            job = cc.write(df, 'new_table',
+                           lnglat=('lng_col', 'lat_col'))
+            while True:
+                curr_status = job.status()['status']
+                if curr_status in ('done', 'failed', 'canceled', 'unknown', ):
+                    print(curr_status)
+                    break
+                time.sleep(5)
+
+    Args:
+        batch_client (carto.sql.BatchSQLClient): Batch SQL API client
+        job (dict): Job status dict returned after sending a Batch SQL API job
+
+    Returns:
+        :obj:`BatchJobStatus`: Instance of BatchJobStatus
+    """
+    def __init__(self, batch_client, job):
+        self.job_id = job['job_id']
+        self.last_status = job['status']
+        self.batch_client = batch_client
+
+    def __repr__(self):
+        return ('BatchJobStatus(job_id=\'{job_id}\', '
+                'last_status=\'{status}\')'.format(job_id=self.job_id,
+                                                   status=self.last_status))
+
+    def status(self):
```

Comment:
@dmed256 you suggested in https://github.com/CartoDB/cartoframes/issues/85 a method for returning dataframes from longer running queries. Here we could have a flag or another method for returning the lnglat'd version of the table/dataframe. I envision us adding more than the lnglat flag for adding geometries.. e.g., street-level geocoding, routing, etc. 

But here, we could have a simple new method (or flag to status), which indicates that when `status=done`, do `cc.read('tablename')` to return the carto'd dataframe.

What do you think? Too soon?

Reply:
I think too soon. I want to scope it out more before including anything else here.